app/src/test/java/app/envelop/ui/login/LoginViewModelTest.kt

```diff
@@ -0,0 +1,78 @@
+package app.envelop.ui.login
+
+import app.envelop.common.Operation
+import app.envelop.domain.LoginService
+import app.envelop.ui.common.Finish
+import app.envelop.ui.common.LoadingState
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import io.reactivex.Single
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+
+class LoginViewModelTest {
+
+    val loginService = mock<LoginService>()
+    val loginViewModel = LoginViewModel()
+
+    @Before
+    fun setUp() {
+        loginViewModel.loginService = loginService
+    }
+
+    @Test
+    fun isLoggingIn() {
+        whenever(loginService.login()).doReturn(Single.just(Operation()))
+
+        val valueStream = loginViewModel.isLoggingIn().test()
+        loginViewModel.loginClick()
+
+        assertEquals(
+            valueStream.values().last(),
+            LoadingState.Idle
+        )
```

Comment:
It could happen in another test, but it would be interesting to test if the value was ever different from Idle after the login click.